lib/vanagon/engine/hardware.rb

```diff
@@ -0,0 +1,102 @@
+require 'vanagon/engine/base'
+require 'json'
+require 'redis'
+
+REDIS_HOST = 'redis.delivery.puppetlabs.net'
+REDIS_PORT = 6379
+LOCK_USER = "vanagon/#{ENV['USER']}"
+
+
+class Vanagon
+  class Engine
+    # Class to use when building on a hardware device (e.g. AIX, Switch, etc)
+    #
+    class Hardware < Base
+
+      # This method is used to obtain a vm to build upon
+      # For the base class we just return the target that was passed in
+      def select_target
+        @build_host
+      end
+
+      # Steps needed to tear down or clean up the system after the build is
+      # complete. In this case, we'll attempt to unlock the hardware
+      def teardown
+        release_lock
+      end
+
+      def initialize(platform)
+        # see if we can aquire a lock
+        Vanagon::Driver.logger.debug "Hardware engine invoked."
+        @platform = platform
+        super
+        @required_attributes << "build_host"
+        @build_host = platform.build_host
+        @target = @build_host
+        aquire_lock
+      end
+
+      # Lock the build_host as specificied in the platform definition. This
+      # implement a rudimentary backoff sequence in order to wait on the host
+      # to unlock. Note: this polls forever.
+      def aquire_lock
+        # Wait if locked
+        if locked?
```

Comment:
This looks like it might do better as a basic loop and not a while loop, as locked? is called twice.

That would look something like...

```
loop do
  if locked?
    ...
  else
    break
  end
end
```